libr/io/io_map.c

```diff
@@ -219,6 +149,8 @@ RIOMap* io_map_new(RIO* io, int fd, int perm, ut64 delta, ut64 addr, ut64 size,
 	r_pvector_push (&io->maps, map);
 	if (do_skyline) {
 		io_map_calculate_skyline (io);
+	} else {
+		add_map_to_skyline (io, map);
```

Comment:
What is the difference between `io_map_calculate_skyline` and this new function? It seems a bit weird to see
```C
if (do_skyline) {
  // compute skyline
} else  {
  // do again something on skyline
}
```

Reply:
@ret2libc `io_map_calculate_skyline()` resets everything and calculates both the skyline and the skyline shadow from scratch, while calling `add_map_to_skyline()` just adds the map to the skyline to make sure we can read from it.
This changes the behavior of `io_map_new()` as before you had to set `do_skyline` to `true` or call `r_io_update()` to read from a newly added map.